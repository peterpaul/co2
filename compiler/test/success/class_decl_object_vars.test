

["stdarg.h"] {
	int printf(char * fmt, ...);
}

class ObjectA
{
	ObjectA() {
		foo = 3;
	}

	int foo;

	int get_foo()
	{
		return foo;
	}
}

class ObjectB
{
	ObjectA data;
	ObjectB(ObjectA data) {
		self.data = data;
	}

	int foo(int bar) {
		return data.get_foo() + bar;
	}
}

int main(int argc, char[][] argv)
{
	ObjectA a = new ObjectA();
	ObjectB b = new ObjectB(a);
	int result = b.foo(6);
	printf("b.foo(9) = %d\n", result);
}
