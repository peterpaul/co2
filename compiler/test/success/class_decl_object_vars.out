b.foo(9) = 9
